test/tests/router_test.js

```diff
@@ -3327,6 +3329,672 @@ test("intermediateTransitionTo() forces an immediate intermediate transition tha
   counterAt(7, "original transition promise resolves");
 });
 
+
+test("Calling transitionTo during initial transition in validation hook should use replaceURL", function(assert) {
+  Transition.currentSequence = 0;
```

Comment:
Tangential to this PR, we should probably store this on the Router instance if possible.